cmd/collect_signals/main.go

```diff
@@ -44,6 +50,71 @@ func init() {
 	}
 }
 
+func handleRepo(ctx context.Context, logger *log.Entry, u *url.URL, out result.Writer) {
+	r, err := projectrepo.Resolve(ctx, u)
+	if err != nil {
+		logger.WithFields(log.Fields{
+			"error": err,
+		}).Warning("Failed to create project")
+		// TODO: we should have an error that indicates that the URL/Project
+		// should be skipped/ignored.
+		return // TODO: add a flag to continue or abort on failure
+	}
+	logger = logger.WithField("canonical_url", r.URL().String())
+
+	// TODO: p.URL() should be checked to see if it has already been processed.
+
+	// Collect the signals for the given project
+	logger.Info("Collecting")
+	ss, err := collector.Collect(ctx, r)
+	if err != nil {
+		var arle *gh.AbuseRateLimitError
+		if errors.As(err, &arle) {
+			logger.WithFields(log.Fields{
+				"status":      arle.Response.Status,
+				"headers":     arle.Response.Header,
+				"message":     arle.Message,
+				"retry_after": arle.RetryAfter,
+			}).Error("Abuse Error error...")
+		}
+		var er *gh.ErrorResponse
+		if errors.As(err, &er) {
+			f, err2 := os.Create("fail.html")
```

Comment:
nit: do we actually need `err2`? We don't need the original `err` in this scope right?

Reply:
This was debugging code. I've removed it.